lib/library.cpp

```diff
@@ -770,7 +770,7 @@ std::string Library::getFunctionName(const Token *ftok, bool *error) const
 
 std::string Library::getFunctionName(const Token *ftok) const
 {
-    if (!Token::Match(ftok, "%name% ("))
+    if (!Token::Match(ftok, "%name% (") && ftok->strAt(-1) != "&")
```

Comment:
hmm.. this looks wrong. I have the feeling `ptr = &foo(x)` takes the address of the return value. I guess you should determine if & is a unary operator and that is easily done with ast.

Reply:
> I have the feeling ptr = &foo(x) takes the address of the return value.

I dont understand how that is relevant. This will get the function name if it is being called with`foo(...)` or its a function pointer like `&foo`(which is the condition I added). There is no change in behavior for `&foo(x)`, just for `&foo`.

> I guess you should determine if & is a unary operator and that is easily done with ast.

How can I do that?

Reply:
> I dont understand how that is relevant

my mistake you are right.

> How can I do that?

Check if the "&" has a `astOperand2()`. If not, it is a unary operator.